fe/fe-core/src/main/java/org/apache/doris/load/GroupCommitManager.java

```diff
@@ -0,0 +1,171 @@
+    private Map<Long, SchemaChangeStatus> statusMap = new HashMap<>();
+
+    public synchronized boolean isBlock(long tableId) {
+        if (statusMap.containsKey(tableId)) {
+            return statusMap.get(tableId) == SchemaChangeStatus.BLOCK;
+        }
+        return false;
+    }
+
+    public synchronized void setStatus(long tableId, SchemaChangeStatus status) {
+        statusMap.put(tableId, status);
+    }
+
+    /**
+     * Check the wal before the endTransactionId is finished or not.
+     */
+    public boolean isPreviousWalFinished(long tableId, long endTransactionId, List<Long> aliveBeIds) {
+        boolean empty = true;
+        for (int i = 0; i < aliveBeIds.size(); i++) {
+            Backend backend = Env.getCurrentSystemInfo().getBackend(aliveBeIds.get(i));
+            PGetWalQueueSizeRequest request = PGetWalQueueSizeRequest.newBuilder()
+                    .setTableId(tableId)
+                    .setTxnId(endTransactionId)
+                    .build();
+            PGetWalQueueSizeResponse response = null;
+            long start = System.currentTimeMillis();
+            boolean done = false;
+            long size = 0;
+            while (!done && System.currentTimeMillis() - start <= Config.check_wal_queue_timeout_threshold) {
+                try {
+                    Future<PGetWalQueueSizeResponse> future = BackendServiceProxy.getInstance()
+                            .getWalQueueSize(new TNetworkAddress(backend.getHost(), backend.getBrpcPort()), request);
+                    response = future.get();
+                } catch (Exception e) {
+                    LOG.warn("encounter exception while getting wal queue size on backend id: " + backend.getId()
+                            + ",exception:" + e);
+                    try {
+                        Thread.sleep(100);
+                    } catch (InterruptedException ie) {
+                        LOG.info("group commit manager sleep wait InterruptedException: ", ie);
+                    }
+                    continue;
+                }
+                TStatusCode code = TStatusCode.findByValue(response.getStatus().getStatusCode());
+                if (code != TStatusCode.OK) {
+                    String msg = "get wal queue size fail,backend id: " + backend.getId() + ", status: "
+                            + response.getStatus();
+                    LOG.warn(msg);
+                    try {
+                        Thread.sleep(100);
+                    } catch (InterruptedException ie) {
+                        LOG.info("group commit manager sleep wait InterruptedException: ", ie);
+                    }
+                    continue;
+                }
+                size = response.getSize();
+                done = true;
+            }
+            if (size > 0) {
+                LOG.info("backend id:" + backend.getId() + ",wal size:" + size);
+                empty = false;
+            }
+        }
+        return empty;
+    }
+
```

Comment:
This method shares a lot of logic with isPreviousWalFinished and could benefit from a common retry mechanism.